lib/node_modules/@stdlib/array/complex64/test/test.fill.js

```diff
@@ -0,0 +1,279 @@
+	t.strictEqual( typeof Complex64Array, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is a `fill` method', function test( t ) {
+	t.strictEqual( hasOwnProp( Complex64Array.prototype, 'fill' ), true, 'has property' );
+	t.strictEqual( isFunction( Complex64Array.prototype.fill ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a complex number array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.fill.call( value );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a complex number', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.fill( value );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a second argument which is not an integer', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.fill( new Complex64( 1.0, 1.0 ), value );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a third argument which is not an integer', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.fill( new Complex64( 1.0, 1.0 ), 0, value );
+		};
+	}
+});
+
+tape( 'the method returns empty array if operating on an empty complex number array', function test( t ) {
+	var arr;
+
+	arr = new Complex64Array();
+	arr.fill( new Complex64( 1.0, 1.0 ) );
+
+	t.strictEqual( arr.length, 0, 'returns expected value' );
```

Comment:
```suggestion
	var arr;
	var out;

	arr = new Complex64Array();
	out = arr.fill( new Complex64( 1.0, 1.0 ) );

	t.strictEqual( instanceOf( out, Complex64Array ), true, 'returns expected value' );
	t.strictEqual( out.length, 0, 'returns expected value' );
```

The test description states "returns an empty array", and thus we should test that the function returns an empty array, not that the original array is empty.